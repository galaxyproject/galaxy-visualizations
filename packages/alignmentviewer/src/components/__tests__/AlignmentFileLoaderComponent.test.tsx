import "jest-webgl-canvas-mock";
import * as React from "react";

import { shallow } from "enzyme";

import { AlignmentFileLoaderComponent } from "../AlignmentFileLoaderComponent";

// Due to the runtime necessities of using styles, we need to explicitly mock out some stub data.
// https://github.com/facebook/jest/issues/3094
jest.mock("../MolecularStyles.module.scss", () => {
  return {
    aaStyBackgroundAlpha_Default: 0.25,
    aaStyDefaultLetterColor_Default: "#ffffff",
    aaStyClass_Default: "mock-aa-class",
    aaStyColorOrder_Default: "",
    aaStyColors_Default: "",
    aaStyBackgroundColorsLightTheme_Default: "",
    aaStyLetterColorsDarkTheme_Default: "",
    aaStyDesc_Default: "mock-aa-style-desc",
    ntStyBackgroundAlpha_Default: 0.25,
    ntStyDefaultLetterColor_Default: "#ffffff",
    ntStyClass_Default: "mock-nt-class",
    ntStyColorOrder_Default: "",
    ntStyColors_Default: "",
    ntStyBackgroundColorsLightTheme_Default: "",
    ntStyLetterColorsDarkTheme_Default: "",
    ntStyDesc_Default: "mock-nt-style-desc",
  };
});

describe("AlignmentFileLoaderComponent", () => {
  it("Should render with minimal props.", () => {
    const wrapper = shallow(
      <AlignmentFileLoaderComponent
        fileSelectorLabelText="text"
        onAlignmentLoaded={() => {}}
        onAlignmenLoadError={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
