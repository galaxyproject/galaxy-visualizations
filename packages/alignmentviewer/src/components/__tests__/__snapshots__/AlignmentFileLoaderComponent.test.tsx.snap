// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignmentFileLoaderComponent Should render with minimal props. 1`] = `
<div
  className="alignment-file-loader"
>
  <div
    className="file-upload-input"
  >
    <label>
      <strong>
        text
      </strong>
      <input
        className="hidden-file-input"
        onChange={[Function]}
        type="file"
      />
      <button
        onClick={[Function]}
        type="button"
      >
        Choose File
      </button>
    </label>
  </div>
  <div
    className="example-files"
  />
</div>
`;
